lib/cext/preprocess.rb

```diff
@@ -246,6 +246,14 @@ def read_write_field(struct_var_name, field_name, leaking)
       {
         match: /if \(stmt_wrapper->refcount == 0\) {/,
         replacement: MYSQL2_FREE_STATEMENT
+      },
+      { 
+        match: /\bparams_enc\[i\](\s*(?:[\),;]|==|!=))/,
+        replacement: 'rb_tr_managed_from_handle(params_enc[i])\1'
+      },
+      {
+        match: /params_enc\[i\](\s*=\s*)(\w.+);\s*$/,
+        replacement: 'params_enc[i]\1rb_tr_handle_for_managed_leaking(\2);'
```

Comment:
Would `read_write_field` work here? I think we'd just need to pass `params_enc[i]` and Regexp.escape `name` in the helper.

Reply:
Could you provide an example of what the read_write_field might look like with this change?  I wasn't sure what you had in mind exactly since it's currently designed for structs with `.` or `->` access.

Reply:
Ah right, I missed those Regexp expect a struct variable on the LHS too.
Then we should probably have a `read_write_array`. I'll leave it to you to add it now or on the next case needing it.

Reply:
I've added `read_write_array`.